File: docs/competitors/typescript/CircleSquare.ts
import * as readline from 'readline'
const readlineInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout
})

const computeCircleArea = (radius: number): string => (Math.PI * radius ** 2).toFixed(3);

const isPositiveNumber = (number: number): boolean => !isNaN(number) && number > 0;

readlineInterface.question("Введите радиус круга: ", (value: string) => {
    const radius: number = parseFloat(value);
    if (isPositiveNumber(radius)) {
        console.log(computeCircleArea(radius))
    } else {
        console.error("Некорректный радиус!")
    }
    readlineInterface.close()
})

File: docs/competitors/python/CircleSquare.py
import math

def computeCircleArea(radius):
    return round(math.pi * radius ** 2, 3)

def isPositiveNumber(number):
    return isinstance(number, (int, float)) and number > 0

value = input("Введите радиус круга: ")

try:
    radius = float(value)
    if isPositiveNumber(radius):
        print(computeCircleArea(radius))
    else:
        print("Некорректный радиус!")
except ValueError:
    print("Некорректный радиус!")
